docs/build/spells.md

```diff
@@ -0,0 +1,197 @@
+
+A spell is a [Marine](/docs/marine-book/introduction.md) service, the main goal of which is to provide state storage for script execution. It also provides some additional features to ease the spell usage for users and take some responsibility from the node (aka [Nox](https://github.com/fluencelabs/nox)). The service serves as a template and is instantiated on the creation of a new spell.
+
+This service is intended to be used as a template and instantiated by a node on each new spell creation.
+
+## Managing spells
+[Nox](https://github.com/fluencelabs/nox) provides user API for interacting with spells.
+
+Users are able to control their spells' lifecycle: install new ones, update them, and remove them. The developer experience is provided by the means of [Fluence CLI](https://github.com/fluencelabs/cli). 
+
+A spell consists of the following parts: 
+- The spell script, which will be executed
+- The list of trigger events (aka the trigger configuration) on which to run the spell
+- The initial state of the spell (e.g. initial state of spell's KV storage)
+
+After installation, the developer is able to update the initial settings: the spell script, the trigger configuration, and the state KV storage.
+
+Also, the developer is able to remove their spells, cleaning the state from the peer.
+
+Developer can check their spells' state: 
+- inspect their internal state (the KV storage, the script, the trigger config)
+- check if they’re currently subscribed to some triggers
+- obtain their peer identity: spell_id and worker_id.
+
+## Events that can trigger spells
+[Nox](https://github.com/fluencelabs/nox) triggers spells according to their configuration. Currently Nox provides the following event triggers:
+- Timer/CRON Triggers: run spells at a specified time or within a specified period.
+Users should be able to specify the following:
+    - when to run the spell,
+    - when to unsubscribe the spell from timer events,
+    - how often should the script be run.
+- Connection Triggers: run spells on connection/disconnection of some peers to the peer.
+
+All other trigger types can be implemented based on the timer triggers & the spell KV storage (blockchain event trigger, database event trigger, HTTP event trigger, etc)
+
+## Creating and updating spells
+
+Spells can be created and managed with [Fluence CLI](https://github.com/fluencelabs/cli). 
+
+Let's start from creating a new project with `fluence init` , which gives us a couple scaffolding choices:
+```bash
+fluence init
+? Select template (Use arrow keys)
+❯ quickstart
+  minimal
+  ts
+  js
+```
+
+Press return to select the default quickstart scaffolding option and enter *quickstart* as the project path when prompted:
+```bash
+? Enter project path or press enter to init in the current directory: quickstart
+Successfully initialized Fluence project template at /Users/arete/Documents/fluence/quickstart
```

Comment:
suggest to use `/tmp` prefix for all paths in all doc pages. It's shorter, and doesn't bring attention to unimportant detail.